packages/nimbus/src/components/search-input/search-input.stories.tsx

```diff
@@ -0,0 +1,785 @@
+
+const meta: Meta<typeof SearchInput> = {
+  title: "components/SearchInput",
+  component: SearchInput,
+};
+
+export default meta;
+
+type Story = StoryObj<typeof SearchInput>;
+
+const inputVariants = ["solid", "ghost"] as const;
+const inputSize = ["md", "sm"] as const;
+
+export const Base: Story = {
+  args: {
+    placeholder: "Search...",
+    ["aria-label"]: "test-search-input",
+  },
+  play: async ({ canvasElement, step }) => {
+    const canvas = within(canvasElement);
+    const input = canvas.getByRole("searchbox");
+
+    await step("Uses a search input element", async () => {
+      await expect(input.tagName).toBe("INPUT");
+      await expect(input).toHaveAttribute("type", "search");
+    });
+
+    await step("Forwards data- & aria-attributes", async () => {
+      await expect(input).toHaveAttribute("aria-label", "test-search-input");
+    });
+
+    await step("Is focusable with <tab> key", async () => {
+      await userEvent.tab();
+      await expect(input).toHaveFocus();
+    });
+
+    await step("Can type text", async () => {
+      await userEvent.type(input, "Search query");
+      await expect(input).toHaveValue("Search query");
+      await userEvent.clear(input);
+    });
+
+    await step("Clear button appears when typing", async () => {
+      await userEvent.type(input, "test");
+      const clearButton = canvas.getByRole("button");
+      await expect(clearButton).toBeInTheDocument();
+      await userEvent.clear(input);
+    });
+  },
+};
+
+export const Sizes: Story = {
+  args: {
+    ["aria-label"]: "test-search-input",
+  },
+  render: (args) => {
+    return (
+      <Stack direction="row" gap="400" alignItems="center">
+        {inputSize.map((size) => (
+          <SearchInput
+            key={size as string}
+            {...args}
+            size={size}
+            placeholder={size as string}
+          />
+        ))}
+      </Stack>
+    );
+  },
+};
+
+export const Variants: Story = {
+  args: {
+    placeholder: "Search...",
+    ["aria-label"]: "test-search-input",
+  },
+  render: (args) => {
+    return (
+      <Stack direction="row" gap="400" alignItems="center">
+        {inputVariants.map((variant) => (
+          <SearchInput
+            key={variant as string}
+            {...args}
+            variant={variant}
+            placeholder={variant as string}
+          />
+        ))}
+      </Stack>
+    );
+  },
+};
+
+export const Disabled: Story = {
+  args: {
+    isDisabled: true,
+  },
+  render: (args) => {
+    return (
+      <Stack direction="row" gap="400" alignItems="center">
+        {inputVariants.map((variant) => (
+          <SearchInput
+            key={variant as string}
+            {...args}
+            variant={variant}
+            placeholder={variant as string}
+            aria-label={`${variant as string}-disabled`}
+          />
+        ))}
+      </Stack>
+    );
+  },
+  play: async ({ canvasElement, step }) => {
```

Comment:
It might be worth making sure that the clear button is also disabled in this test

Reply:
Thanks, see 546328c097b4d9bcd8ead3292d9e0028aefb6e2f